### PR TITLE
Create Rule “do-you-use-co-creation-patterns/rule”

### PR DESCRIPTION
**Pull request by danielmackay**
Automatically generated by Netlify CMS

### CODE REVIEW

File: rules/do-you-use-co-creation-patterns/rule.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+authors:
+  - title: Daniel Mackay
+    url: https://ssw.com.au/people/daniel-mackay
+related: []
+created: 2022-11-21T00:28:43.256Z
+guid: 4ff555fd-3f4c-416b-9fc3-1bf9409cc0ab
+---
+These days pull requests are the de facto standard for getting code reviewed.  Once a developer has finished their change, they will typically submit a pull request and move on to their next task.  This allows for an asynchronous process to take place which may seem like a good idea, but this can also lead to inefficiencies.
+
+## Inefficient Code Reviews
```
Comment (pierssinclairssw) [inline rules/do-you-use-co-creation-patterns/rule.md:14]: Change heading to "Problem - Inefficient Code Reviews"

Then I would summarize the sub headings as bullet points to give the reader an idea of what they are in for

e.g.
Pull Requests can cause inefficient code review processes due to:
* Providing feedback too late
* Providing feedback too slow
* ...
* ...

File: rules/do-you-use-co-creation-patterns/rule.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+When a developer submits a change for review, they will often move onto a new task and context switch.  After their change has been reviewed, they need to switch from the current task back to the old task, address feedback, then switch from the old task to the current task.  The more pull requests a developer has open the more context switching needs to take place.  This division of focus can lead to lower quality code.
+
+### Too Much Work in Progress
+
+One of the DORA Metrics used to measure high-performing teams is called [Lead Time for Change](https://www.leanix.net/en/wiki/vsm/dora-metrics#lead-time-for-changes).  High-performing teams can get changes into production faster.  
+
+The more work in progress a developer has, the more context switching is required and the longer it will take to get these changes 'done'.  This also has a flow on effect of the more work in progress a developer has, the longer it will take them to review another developers code,
+
+The more WIP we have in the system, the less responsive it becomes, and the longer the wait times and delays, which in turn incentivizes even higher WIP because people pull in more stuff as they need to wait.
```
Comment (pierssinclairssw) [inline rules/do-you-use-co-creation-patterns/rule.md:44]: I think a few of these points can be combined into 1 so the problems are easier to digest

Too much work in progress, excessive context switching and providing feedback too slow can probably be one point, and just make sure to cover all the same parts.

**Review by pierssinclairssw — CHANGES_REQUESTED** [review]

**Review by jernejk — APPROVED** [review]
LGTM.

Just adding a comment that large pull requests, while not great, can help some devs get from start to finish without losing sight of what they need to do. Splitting up into smaller PR makes it harder to get a full picture of the change and figure out what else needs to be done.

With pair/mob programming, we need to take into consideration that it's great for building things together but can cause the person not to be able to focus on solving the problem as it can introduce more distractions. I believe that alone time is as necessary as pair programming.

File: rules/do-you-use-co-creation-patterns/rule.md (hunk 1 above)
Comment (danielmackay) [reply rules/do-you-use-co-creation-patterns/rule.md]: Updated

File: rules/do-you-use-co-creation-patterns/rule.md (hunk 2 above)
Comment (danielmackay) [reply rules/do-you-use-co-creation-patterns/rule.md]: No longer relevant due to recent change

File: rules/do-you-use-co-creation-patterns/rule.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+### Create Small PRs
+
+Advantages of small pull requests:
+
+1. Require a smaller block of time to review - this makes it easier for the reviewer to find the time
+2. Get feedback earlier - makes change easier to adopt
+3. Less risk - reduces the chance of an incorrect approach being taken
+4. Get quality feedback - small blocks of code are easier to digest
+
+## Co-Creation Patterns
```
[the remark this answers is not in the document]
Comment (danielmackay) [reply rules/do-you-use-co-creation-patterns/rule.md]: Done

File: rules/do-you-use-co-creation-patterns/rule.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+Asynchronous feedback can sometimes not be clear.  This can result in back-and-forth discussing the feedback which results in more waste and more context switching.
+
+<!--StartFragment-->
+![](https://imgopt.infoq.com/fit-in/1200x2400/filters:quality(80)/filters:no_upscale()/articles/co-creation-patterns-software-development/en/resources/61-1667592466401.jpeg)
+<!--EndFragment-->
+
+## Efficient Code Reviews
+
+There are several improvements we can make to address some of the issues above.
```
[the remark this answers is not in the document]
Comment (danielmackay) [reply rules/do-you-use-co-creation-patterns/rule.md:35]: Done

File: rules/do-you-use-co-creation-patterns/rule.md [hunk 5]
```diff
@@ -60,14 +60,14 @@ We can avoid this by using co-creation patterns.
 
 > As a general rule pull requests with less that 20 lines of code and larger changes with a degree of complexity or risk make good candidates for co-creation
 
-### Techniques
+### Patterns
```
Comment (jernejk) [inline rules/do-you-use-co-creation-patterns/rule.md:61]: Is the way we work a pattern?
I would think this is more like a best practice or something like that.

Comment (danielmackay) [reply rules/do-you-use-co-creation-patterns/rule.md:61]: I changed due to the rule being called 'co-creation patterns' yet, I hadn't referenced any patterns anywhere.

**Comment by danielmackay** [comment]
@pierssinclairssw - what categories should I attach this rule to?

File: rules/do-you-use-co-creation-patterns/rule.md (hunk 5 above)
Comment (william-liebenberg) [reply rules/do-you-use-co-creation-patterns/rule.md:61]: I'd stick with patterns... as in "patterns of behaviour" makes sense for how can work together in different ways :)

**Review by william-liebenberg — APPROVED** [review]
LGTM

**Pull request merged**